owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt

```diff
@@ -190,25 +188,24 @@ class SettingsSecurityFragment : PreferenceFragmentCompat() {
             }
         }
 
-        // Lock application and Access from document provider visibility
+        // Lock application
         if (prefPasscode?.isChecked == false && prefPattern?.isChecked == false) {
             prefLockApplication?.isEnabled = false
-            disableAccessFromDocumentProvider()
         }
 
-        // Access from document provider
-        prefAccessDocumentProvider?.setOnPreferenceChangeListener { preference: Preference?, newValue: Any ->
-            if (newValue as Boolean) {
+        // Lock access from document provider
+        prefLockAccessDocumentProvider?.setOnPreferenceChangeListener { _: Preference?, newValue: Any ->
+            if (!(newValue as Boolean)) {
```

Comment:
I don't know if this is the expected behavior 🤔  

Not sure if we should show a dialog to the user when he wants to enable the documents provider(which is the expected behavior) from a UX point of view

Reply:
IMHO, the dialog does not make sense here since access to document provider is open by default. Warning users when the `Lock` option is switched off would be the same as warning when the `Security` section is opened. I'd remove the dialog because the feature meaning is opposite as the previous one.

Reply:
Ok, I will remove it then. Thank you!